pkg/sql/lease.go

```diff
@@ -777,30 +800,64 @@ func (t *tableState) acquireWait() {
 // inserted at the end of the lease set (i.e. it will be returned by
 // findNewest() from now on).
 //
+// acquireNodeLease begins an asynchronous process to acquire a lease.
+// Use errChan to receive any errors from the asynchronous process. If
+// error is not nil, expect the error channel to be nil.
+//
+// Use acquireWait to block until the lease is acquired.
+//
 // t.mu needs to be locked.
 func (t *tableState) acquireNodeLease(
 	ctx context.Context, m *LeaseManager, minExpirationTime hlc.Timestamp,
-) error {
+) (chan error, error) {
 	if m.isDraining() {
-		return errors.New("cannot acquire lease when draining")
+		return nil, errors.New("cannot acquire lease when draining")
 	}
 
-	// Notify when lease has been acquired.
+	// Notify when lease is being acquired.
 	t.mu.acquiring = make(chan struct{})
+	errChan := make(chan error)
```

Comment:
create this channel at the top and use it throughout. Use a buffer of 1 so that doesn't block on send or receive